### PR TITLE
chore: remove support for Kubernetes 1.15

### LINKED ISSUE #3631: Deprecate Kubernetes v1.15 support
A tracking issue to remove support for building new v1.15 clusters.

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

This PR removes support for creating new clusters with Kubernetes 1.15.

**Issue Fixed**:

Fixes #3631

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/test_cluster_configs/network_policy/cilium.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
 		"GINKGO_FOCUS": "should apply various network policies"
 	},
 	"options": {
-		"allowedOrchestratorVersions": ["1.15"]
+		"allowedOrchestratorVersions": []
```
Comment (mboersma) [inline test/e2e/test_cluster_configs/network_policy/cilium.json:6]: Hmm...

**Comment by jadarsie** [comment]
A bunch of these `{{- if IsKubernetesVersionGe "1.16.0"}}` can be removed from templates under `parts/k8s`.

Example: https://github.com/Azure/aks-engine/blob/master/parts/k8s/addons/kube-proxy.yaml#L51-L54

**Comment by mboersma** [comment]
> `{{- if IsKubernetesVersionGe "1.16.0"}}` can be removed from the templates

You are so right! Thanks, I'll make those changes and update this PR.

File: test/e2e/test_cluster_configs/network_policy/cilium.json (hunk 1 above)
Comment (jackfrancis) [reply test/e2e/test_cluster_configs/network_policy/cilium.json:6]: Will delete this test config entirely

**Review by jackfrancis — DISMISSED** [review]
/lgtm

**Pull request merged**